packages/admin-panel/src/sync/SyncStatus.js

```diff
@@ -0,0 +1,214 @@
+`;
+
+const StatusMessage = styled.div``;
+
+const spin = keyframes`
+  0% { 
+      transform: rotate(360deg); 
+  }
+
+   100% { 
+      transform: rotate(0deg); 
+  }
+ `;
+
+const SyncSuccessIcon = styled(CheckCircleIcon)`
+  color: ${props => props.theme.palette.success.main};
+`;
+
+const SyncFailingIcon = styled(ErrorIcon)`
+  color: ${props => props.theme.palette.error.main};
+`;
+
+const SpinningSyncIcon = styled(SyncIcon)`
+  color: white;
+
+  animation: 3s ${spin};
+  animation-timing-function: linear;
+  animation-iteration-count: infinite;
+`;
+
+const SyncingIconButton = styled(IconButton)`
+  display: flex;
+
+  background-color: ${props => props.theme.palette.blue[100]};
+
+  &.Mui-disabled {
+    background-color: ${props => props.theme.palette.primary.main};
+    color: white;
+  }
+`;
+
+const IdleIconButton = styled(IconButton)``;
+
+// Bit of a hack to work around the fact that the sync button is constantly being recreated
+// in the resource page due to parent component re-rendering https://stackoverflow.com/a/33800398
+const externalState = {};
+
+const useExternalState = (key, initialState) => {
+  const [state, setState] = useState(() => {
+    if (key in externalState) {
+      return externalState[key];
+    }
+    return initialState;
+  });
+
+  const onChange = nextState => {
+    externalState[key] = nextState;
+    setState(nextState);
+  };
+
+  return [state, onChange];
+};
+
+const formatLog = ({ timestamp, message }) =>
+  `${formatDistance(new Date(timestamp.concat(' UTC')), new Date(), {
+    addSuffix: true,
+  })}: ${message}`;
+
+export const SyncStatus = props => {
+  const { actionConfig, original } = props;
+  const api = useApi();
+  const [status, setStatus] = useExternalState(`${original.id}.status`, original.sync_status);
+  const [logMessage, setLogMessage] = useExternalState(`${original.id}.logMessage`, '');
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const syncStatusEndpoint = makeSubstitutionsInString(actionConfig.syncStatusEndpoint, original);
+  const latestSyncLogEndpoint = makeSubstitutionsInString(
+    actionConfig.latestSyncLogEndpoint,
+    original,
+  );
+  const manualSyncEndpoint = makeSubstitutionsInString(actionConfig.manualSyncEndpoint, original);
+
+  const pollStatus = async () => {
+    try {
+      const statusResponse = await api.get(syncStatusEndpoint);
+      const latestLogResponse = await api.get(latestSyncLogEndpoint);
+      setStatus(statusResponse.body.sync_status);
+      setLogMessage(formatLog(latestLogResponse.body.logs[0]));
+      setErrorMessage(null);
+    } catch (error) {
+      setErrorMessage(error.message);
+    }
+  };
+
+  // First poll after 0.5 seconds, then poll each 5 seconds
+  useEffect(() => {
+    const timeout = setTimeout(pollStatus, 500);
+    return () => clearTimeout(timeout);
+  }, []);
+
+  useEffect(() => {
+    const timer = setInterval(pollStatus, 5000);
+    return () => clearInterval(timer);
+  }, []);
```

Comment:
@rohan-bes Do you think we should stop polling the sync log while user is not focus on the page?

Reply:
So I believe it won't poll when a user goes to a page that isn't the Sync Groups page. However it will keep polling while the logs modal is up for instance. I feel that's okay, what do you think? Just in the spirit of keeping things simple

Reply:
I've seen it keep polling on the sync groups page. My only concern is it could make the log super long if someone didn't close the page. It is not a huge problem so it is optional.

Reply:
I might bump the poll time up to 10 seconds.

I think polling inevitably causes a bit of noise, just the nature of things. A websockets solution is really the way to go for a dynamic front-end, but that's a lot more work 😓